src/runtime_src/hip/core/common.h

```diff
@@ -13,17 +18,26 @@ namespace xrt::core::hip {
 struct ctx_info
 {
   context_handle ctx_hdl{nullptr};
-  device_handle dev_hdl{std::numeric_limits<uint32_t>::max()};
   bool active{false};
 };
 
 // thread local hip objects
 struct hip_tls_objs
 {
+  device_handle dev_hdl{std::numeric_limits<uint32_t>::max()};
   std::stack<std::weak_ptr<context>> ctx_stack;
   ctx_info pri_ctx_info;
 };
 extern thread_local hip_tls_objs tls_objs;
 } // xrt::core::hip
 
+namespace {
+// common function that throws hip error
+inline void
+throw_if(bool check, hipError_t err, const std::string& err_msg)
+{
+  check ? throw xrt_core::system_error(err, err_msg) : void();
```

Comment:
Nice! But that looks a little bit obfuscated
```suggestion
  if (check)
    throw xrt_core::system_error(err, err_msg);
```
Perhaps you could introduce some `throw_invalid_value_if` and so on for common cases?

Reply:
@keryell I have added changes as suggested, please review